File: 0001_two-sum/python3/solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, val in enumerate(nums):
            complement = target - val
            if complement in seen:
                return [seen[complement], i]
            seen[val] = i
        return []

File: 0001_two-sum/rust/solution.rs
use std::collections::HashMap;

#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        for (i, &val) in nums.iter().enumerate() {
            let complement = target - val;
            if let Some(&j) = map.get(&complement) {
                return vec![j as i32, i as i32];
            }
            map.insert(val, i);
        }
        vec![]
    }
}
